target/linux/ramips/dts/DWR-512-B.dts

```diff
@@ -0,0 +1,115 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "ralink,rt5350-soc";
+	model = "D-Link DWR-512 B";
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 0 1>;
+			linux,code = <0x211>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sms {
+			label = "dwr-512-b:green:sms";
+			gpios = <&gpio0 8 1>;
+		};
+		status {
+			label = "dwr-512-b:green:status";
+			gpios = <&gpio0 9 1>;
+		};
+		2g {
+			label = "dwr-512-b:green:2g";
+			gpios = <&gpio0 17 1>;
+		};
+		3g {
+			label = "dwr-512-b:green:3g";
+			gpios = <&gpio0 19 1>;
+		};
+		sstrengthr {
+			label = "dwr-512-b:red:sigstrength";
+			gpios = <&gpio0 20 1>;
+		};
+		sstrengthg {
+			label = "dwr-512-b:green:sigstrength";
+			gpios = <&gpio0 21 1>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+ 
+		modem3g_enable {
+			gpio-export,name = "modem3g_enable";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 18 1>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	mx25l6405d@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25l6405d", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		fast-read;
+
+                partition@0 {
+                        label = "u-boot";
+                        reg = <0x0 0x30000>;
+                        read-only;
+                };
+
+                partition@30000 {
+                        label = "u-boot-env";
+                        reg = <0x30000 0x10000>;
+                        read-only;
+                };
+
+                factory: partition@40000 {
+                        label = "factory";
+                        reg = <0x40000 0x10000>;
+                        read-only;
+                };
+
+                partition@50000 {
+                        label = "firmware";
+                        reg = <0x50000 0x7a0000>;
+                };
+
+		config: partition@7f0000 {
+			label = "config";
+			reg = <0x7f0000 0x10000>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "jtag", "uartf";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&config 0xe07a>;
```

Comment:
You should look for some hex values where 5053 + 1814 or 5350 + 1418 are nearby. Your wireless mac address should be between these two values. 1814 is the ralink (pci) vendor id. These values can be found in all EEPROMs dump I've pasted.

But I might be wrong on this and it's different for RT5350 or this particular board. Would it be possible that you upload a dump of the full flash at the time the D-Link firmware was installed (before you made any changes).